src/main/java/gregtech/client/particle/GTParticle.java

```diff
@@ -1,191 +1,124 @@
 package gregtech.client.particle;
 
-import net.minecraft.client.particle.Particle;
+import gregtech.client.renderer.IRenderSetup;
 import net.minecraft.client.renderer.BufferBuilder;
 import net.minecraft.entity.Entity;
-import net.minecraft.world.World;
+import net.minecraft.util.math.Vec3d;
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
-import java.util.function.Consumer;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
 
 /**
- * Created with IntelliJ IDEA.
- *
- * @Author: KilaBash
- * @Date: 2021/08/31
- * @Description:
+ * A custom particle implementation with framework for more advanced rendering capabilities.<p/>
+ * GTParticle instances are managed by {@link GTParticleManager}. GTParticle instances with same {@link IRenderSetup}s
```

Comment:
Would be a good idea that this means they must have the same `equals()` and `hashCode()` return values for the batch drawing.